diff -ruN a/config/kernel-vm_node_stat.m4 b/config/kernel-vm_node_stat.m4
--- a/config/kernel-vm_node_stat.m4	2018-01-20 16:59:03.199311613 +0000
+++ b/config/kernel-vm_node_stat.m4	2018-01-20 17:00:39.245615527 +0000
@@ -10,7 +10,7 @@
         ],[
 			int a __attribute__ ((unused)) = NR_VM_NODE_STAT_ITEMS;
 			long x __attribute__ ((unused)) =
-				atomic_long_read(&vm_node_stat[0]);
+				atomic_long_read_unchecked(&vm_node_stat[0]);
 			(void) global_node_page_state(0);
 	],[
 		AC_MSG_RESULT(yes)
