diff -ruN a/out/mkcodecache.target.mk b/out/mkcodecache.target.mk
--- a/out/mkcodecache.target.mk	2020-06-16 21:02:03.373831461 +0000
+++ b/out/mkcodecache.target.mk	2020-06-16 21:04:10.222897014 +0000
@@ -162,6 +162,7 @@
 $(builddir)/mkcodecache: TOOLSET := $(TOOLSET)
 $(builddir)/mkcodecache: $(OBJS) $(obj).target/libnode.a $(obj).target/deps/histogram/libhistogram.a $(obj).target/deps/uvwasi/libuvwasi.a $(obj).target/tools/v8_gypfiles/libv8_libplatform.a $(obj).target/deps/llhttp/libllhttp.a $(obj).target/deps/brotli/libbrotli.a $(obj).target/deps/uv/libuv.a $(obj).target/tools/v8_gypfiles/libv8_base_without_compiler.a $(obj).target/tools/v8_gypfiles/libv8_libbase.a $(obj).target/tools/v8_gypfiles/libv8_libsampler.a $(obj).target/tools/v8_gypfiles/libv8_compiler.a $(obj).target/tools/v8_gypfiles/libv8_snapshot.a $(obj).target/tools/v8_gypfiles/libv8_initializers.a FORCE_DO_CMD
 	$(call do_cmd,link)
+	paxctl-ng -m $(builddir)/mkcodecache
 
 all_deps += $(builddir)/mkcodecache
 # Add target alias
diff -ruN a/out/node_mksnapshot.target.mk b/out/node_mksnapshot.target.mk
--- a/out/node_mksnapshot.target.mk	2020-06-16 21:02:03.373831461 +0000
+++ b/out/node_mksnapshot.target.mk	2020-06-16 21:03:19.212732027 +0000
@@ -162,6 +162,7 @@
 $(builddir)/node_mksnapshot: TOOLSET := $(TOOLSET)
 $(builddir)/node_mksnapshot: $(OBJS) $(obj).target/libnode.a $(obj).target/deps/histogram/libhistogram.a $(obj).target/deps/uvwasi/libuvwasi.a $(obj).target/tools/v8_gypfiles/libv8_libplatform.a $(obj).target/deps/llhttp/libllhttp.a $(obj).target/deps/brotli/libbrotli.a $(obj).target/deps/uv/libuv.a $(obj).target/tools/v8_gypfiles/libv8_base_without_compiler.a $(obj).target/tools/v8_gypfiles/libv8_libbase.a $(obj).target/tools/v8_gypfiles/libv8_libsampler.a $(obj).target/tools/v8_gypfiles/libv8_compiler.a $(obj).target/tools/v8_gypfiles/libv8_snapshot.a $(obj).target/tools/v8_gypfiles/libv8_initializers.a FORCE_DO_CMD
 	$(call do_cmd,link)
+	paxctl-ng -m $(builddir)/node_mksnapshot
 
 all_deps += $(builddir)/node_mksnapshot
 # Add target alias
diff -ruN a/out/tools/v8_gypfiles/mksnapshot.target.mk b/out/tools/v8_gypfiles/mksnapshot.target.mk
--- a/out/tools/v8_gypfiles/mksnapshot.target.mk	2020-06-16 21:02:03.377833514 +0000
+++ b/out/tools/v8_gypfiles/mksnapshot.target.mk	2020-06-16 21:02:37.331249472 +0000
@@ -173,6 +173,7 @@
 $(builddir)/mksnapshot: TOOLSET := $(TOOLSET)
 $(builddir)/mksnapshot: $(OBJS) $(obj).target/tools/v8_gypfiles/libv8_base_without_compiler.a $(obj).target/tools/v8_gypfiles/libv8_init.a $(obj).target/tools/v8_gypfiles/libv8_libbase.a $(obj).target/tools/v8_gypfiles/libv8_libplatform.a $(obj).target/tools/v8_gypfiles/libv8_nosnapshot.a $(obj).target/tools/v8_gypfiles/libv8_libsampler.a $(obj).target/tools/v8_gypfiles/libv8_compiler.a $(obj).target/tools/v8_gypfiles/libv8_initializers.a FORCE_DO_CMD
 	$(call do_cmd,link)
+	paxctl-ng -m $(builddir)/mksnapshot
 
 all_deps += $(builddir)/mksnapshot
 # Add target alias
