diff -ruN a/tools/v8_context_snapshot/BUILD.gn b/tools/v8_context_snapshot/BUILD.gn
--- a/tools/v8_context_snapshot/BUILD.gn	2018-07-18 19:09:07.000000000 +0000
+++ b/tools/v8_context_snapshot/BUILD.gn	2018-07-20 14:00:08.530129742 +0000
@@ -56,6 +56,19 @@
     }
   }
 
+  action("pax_mark_v8_context_snapshot_generator") {
+    script = "run.py"
+    sources = []
+    outputs = [ "$target_out_dir/generator_is_pax_marked" ]
+    data = []
+    args = [
+      "pax-mark-m.sh",
+      "./" + rebase_path(get_label_info(":v8_context_snapshot_generator($v8_snapshot_toolchain)",
+                                        "root_out_dir") + "/v8_context_snapshot_generator"),
+      "$target_out_dir/generator_is_pax_marked"
+    ]
+  }
+
   action("generate_v8_context_snapshot") {
     script = "//build/gn_run_binary.py"
     output_file = "$root_out_dir/v8_context_snapshot.bin"
@@ -76,6 +87,7 @@
 
     deps = [
       ":v8_context_snapshot_generator($v8_snapshot_toolchain)",
+      ":pax_mark_v8_context_snapshot_generator"
     ]
   }
 
diff -ruN a/tools/v8_context_snapshot/pax-mark-m.sh b/tools/v8_context_snapshot/pax-mark-m.sh
--- a/tools/v8_context_snapshot/pax-mark-m.sh	1970-01-01 00:00:00.000000000 +0000
+++ b/tools/v8_context_snapshot/pax-mark-m.sh	2018-07-20 13:43:17.170116782 +0000
@@ -0,0 +1,8 @@
+#!/bin/bash
+
+set -e
+set -v
+source "${EPREFIX}/usr/portage/eclass/pax-utils.eclass"
+
+pax-mark m "${1}"
+touch "${2}"
diff -ruN a/tools/v8_context_snapshot/run.py b/tools/v8_context_snapshot/run.py
--- a/tools/v8_context_snapshot/run.py	1970-01-01 00:00:00.000000000 +0000
+++ b/tools/v8_context_snapshot/run.py	2018-07-20 13:43:17.170116782 +0000
@@ -0,0 +1,12 @@
+#!/usr/bin/env python
+# Copyright 2014 the V8 project authors. All rights reserved.
+# Use of this source code is governed by a BSD-style license that can be
+# found in the LICENSE file.
+
+"""This program wraps an arbitrary command since gn currently can only execute
+scripts."""
+
+import subprocess
+import sys
+
+sys.exit(subprocess.call(sys.argv[1:]))
diff -ruN a/v8/BUILD.gn b/v8/BUILD.gn
--- a/v8/BUILD.gn	2018-07-18 19:10:23.000000000 +0000
+++ b/v8/BUILD.gn	2018-07-20 13:51:42.380169724 +0000
@@ -937,6 +937,19 @@
   configs = [ ":internal_config" ]
 }
 
+action("pax_mark_mksnapshot") {
+  script = "tools/run.py"
+  sources = []
+  outputs = [ "$target_out_dir/mksnapshot_is_pax_marked" ]
+  data = []
+  args = [
+    "tools/pax-mark-m.sh",
+    "./" + rebase_path(get_label_info(":mksnapshot($v8_snapshot_toolchain)",
+                                      "root_out_dir") + "/mksnapshot"),
+    "$target_out_dir/mksnapshot_is_pax_marked"
+  ]
+}
+
 # Template to generate different V8 snapshots based on different runtime flags.
 # Can be invoked with run_mksnapshot(<name>). The target will resolve to
 # run_mksnapshot_<name>. If <name> is "default", no file suffixes will be used.
@@ -959,6 +970,7 @@
 
     deps = [
       ":mksnapshot($v8_snapshot_toolchain)",
+      ":pax_mark_mksnapshot"
     ]
 
     script = "tools/run.py"
diff -ruN a/v8/tools/pax-mark-m.sh b/v8/tools/pax-mark-m.sh
--- a/v8/tools/pax-mark-m.sh	1970-01-01 00:00:00.000000000 +0000
+++ b/v8/tools/pax-mark-m.sh	2018-07-20 13:43:17.170116782 +0000
@@ -0,0 +1,8 @@
+#!/bin/bash
+
+set -e
+set -v
+source "${EPREFIX}/usr/portage/eclass/pax-utils.eclass"
+
+pax-mark m "${1}"
+touch "${2}"
