diff -ruN a/config/kernel-group-info.m4 b/config/kernel-group-info.m4
--- a/config/kernel-group-info.m4	2018-03-27 11:22:56.914757543 +0000
+++ b/config/kernel-group-info.m4	2018-03-27 11:24:53.218556237 +0000
@@ -5,7 +5,7 @@
 AC_DEFUN([SPL_AC_GROUP_INFO_GID], [
 	AC_MSG_CHECKING([whether group_info->gid exists])
 	tmp_flags="$EXTRA_KCFLAGS"
-	EXTRA_KCFLAGS="-Werror"
+	EXTRA_KCFLAGS=""
 	SPL_LINUX_TRY_COMPILE([
 		#include <linux/cred.h>
 	],[
diff -ruN a/config/kernel-inode-lock.m4 b/config/kernel-inode-lock.m4
--- a/config/kernel-inode-lock.m4	2018-03-27 11:22:56.914757543 +0000
+++ b/config/kernel-inode-lock.m4	2018-03-27 11:24:53.218556237 +0000
@@ -7,7 +7,7 @@
 AC_DEFUN([SPL_AC_INODE_LOCK], [
 	AC_MSG_CHECKING([whether inode_lock_shared() exists])
 	tmp_flags="$EXTRA_KCFLAGS"
-	EXTRA_KCFLAGS="-Werror"
+	EXTRA_KCFLAGS=""
 	SPL_LINUX_TRY_COMPILE([
 		#include <linux/fs.h>
 	],[
diff -ruN a/config/kernel-kmem-cache.m4 b/config/kernel-kmem-cache.m4
--- a/config/kernel-kmem-cache.m4	2018-03-27 11:22:56.914757543 +0000
+++ b/config/kernel-kmem-cache.m4	2018-03-27 11:24:53.218556237 +0000
@@ -43,7 +43,7 @@
 AC_DEFUN([SPL_AC_KMEM_CACHE_CREATE_USERCOPY], [
 	AC_MSG_CHECKING([whether kmem_cache_create_usercopy() exists])
 	tmp_flags="$EXTRA_KCFLAGS"
-	EXTRA_KCFLAGS="-Werror"
+	EXTRA_KCFLAGS=""
 	SPL_LINUX_TRY_COMPILE([
 		#include <linux/slab.h>
 		static void ctor(void *foo)
diff -ruN a/config/kernel-rw.m4 b/config/kernel-rw.m4
--- a/config/kernel-rw.m4	2018-03-27 11:22:56.914757543 +0000
+++ b/config/kernel-rw.m4	2018-03-27 11:24:53.218556237 +0000
@@ -6,7 +6,7 @@
 AC_DEFUN([SPL_AC_KERNEL_WRITE], [
 	AC_MSG_CHECKING([whether kernel_write() takes loff_t pointer])
 	tmp_flags="$EXTRA_KCFLAGS"
-	EXTRA_KCFLAGS="-Werror"
+	EXTRA_KCFLAGS=""
 	SPL_LINUX_TRY_COMPILE([
 		#include <linux/fs.h>
 	],[
@@ -35,7 +35,7 @@
 AC_DEFUN([SPL_AC_KERNEL_READ], [
 	AC_MSG_CHECKING([whether kernel_read() takes loff_t pointer])
 	tmp_flags="$EXTRA_KCFLAGS"
-	EXTRA_KCFLAGS="-Werror"
+	EXTRA_KCFLAGS=""
 	SPL_LINUX_TRY_COMPILE([
 		#include <linux/fs.h>
 	],[
diff -ruN a/config/kernel-rwsem.m4 b/config/kernel-rwsem.m4
--- a/config/kernel-rwsem.m4	2018-03-27 11:22:56.914757543 +0000
+++ b/config/kernel-rwsem.m4	2018-03-27 11:24:53.222556230 +0000
@@ -7,7 +7,7 @@
 AC_DEFUN([SPL_AC_RWSEM_SPINLOCK_IS_RAW], [
 	AC_MSG_CHECKING([whether struct rw_semaphore member wait_lock is raw])
 	tmp_flags="$EXTRA_KCFLAGS"
-	EXTRA_KCFLAGS="-Werror"
+	EXTRA_KCFLAGS=""
 	SPL_LINUX_TRY_COMPILE([
 		#include <linux/rwsem.h>
 	],[
@@ -33,7 +33,7 @@
 AC_DEFUN([SPL_AC_RWSEM_ACTIVITY], [
 	AC_MSG_CHECKING([whether struct rw_semaphore has member activity])
 	tmp_flags="$EXTRA_KCFLAGS"
-	EXTRA_KCFLAGS="-Werror"
+	EXTRA_KCFLAGS=""
 	SPL_LINUX_TRY_COMPILE([
 		#include <linux/rwsem.h>
 	],[
@@ -58,7 +58,7 @@
 	AC_MSG_CHECKING(
 	[whether struct rw_semaphore has atomic_long_t member count])
 	tmp_flags="$EXTRA_KCFLAGS"
-	EXTRA_KCFLAGS="-Werror"
+	EXTRA_KCFLAGS=""
 	SPL_LINUX_TRY_COMPILE([
 		#include <linux/rwsem.h>
 	],[
diff -ruN a/config/kernel-set-fs-pwd.m4 b/config/kernel-set-fs-pwd.m4
--- a/config/kernel-set-fs-pwd.m4	2018-03-27 11:22:56.914757543 +0000
+++ b/config/kernel-set-fs-pwd.m4	2018-03-27 11:24:53.222556230 +0000
@@ -4,7 +4,7 @@
 dnl #
 AC_DEFUN([SPL_AC_SET_FS_PWD_WITH_CONST],
 	tmp_flags="$EXTRA_KCFLAGS"
-	EXTRA_KCFLAGS="-Werror"
+	EXTRA_KCFLAGS=""
 	[AC_MSG_CHECKING([whether set_fs_pwd() requires const struct path *])
 	SPL_LINUX_TRY_COMPILE([
 		#include <linux/spinlock.h>
diff -ruN a/config/kernel-shrinker.m4 b/config/kernel-shrinker.m4
--- a/config/kernel-shrinker.m4	2018-03-27 11:22:56.914757543 +0000
+++ b/config/kernel-shrinker.m4	2018-03-27 11:24:53.222556230 +0000
@@ -1,6 +1,6 @@
 AC_DEFUN([SPL_AC_SHRINKER_CALLBACK],[
 	tmp_flags="$EXTRA_KCFLAGS"
-	EXTRA_KCFLAGS="-Werror"
+	EXTRA_KCFLAGS=""
 	dnl #
 	dnl # 2.6.23 to 2.6.34 API change
 	dnl # ->shrink(int nr_to_scan, gfp_t gfp_mask)
diff -ruN a/config/kernel-spinlock.m4 b/config/kernel-spinlock.m4
--- a/config/kernel-spinlock.m4	2018-03-27 11:22:56.914757543 +0000
+++ b/config/kernel-spinlock.m4	2018-03-27 11:24:53.222556230 +0000
@@ -6,7 +6,7 @@
 AC_DEFUN([SPL_AC_FS_STRUCT_SPINLOCK], [
 	AC_MSG_CHECKING([whether struct fs_struct uses spinlock_t])
 	tmp_flags="$EXTRA_KCFLAGS"
-	EXTRA_KCFLAGS="-Werror"
+	EXTRA_KCFLAGS=""
 	SPL_LINUX_TRY_COMPILE([
 		#include <linux/sched.h>
 		#include <linux/fs_struct.h>
diff -ruN a/config/kernel-timer.m4 b/config/kernel-timer.m4
--- a/config/kernel-timer.m4	2018-03-27 11:22:56.914757543 +0000
+++ b/config/kernel-timer.m4	2018-03-27 11:24:53.226556223 +0000
@@ -9,7 +9,7 @@
 AC_DEFUN([SPL_AC_KERNEL_TIMER_FUNCTION_TIMER_LIST], [
 	AC_MSG_CHECKING([whether timer_list.function gets a timer_list])
 	tmp_flags="$EXTRA_KCFLAGS"
-	EXTRA_KCFLAGS="-Werror"
+	EXTRA_KCFLAGS=""
 	SPL_LINUX_TRY_COMPILE([
 		#include <linux/timer.h>
 		void task_expire(struct timer_list *tl) {}
diff -ruN a/config/spl-build.m4 b/config/spl-build.m4
--- a/config/spl-build.m4	2018-03-27 11:22:56.914757543 +0000
+++ b/config/spl-build.m4	2018-03-27 11:24:53.226556223 +0000
@@ -448,8 +448,8 @@
 
 	AS_IF([test "x$enable_debug" = xyes],
 	[
-		KERNELCPPFLAGS="${KERNELCPPFLAGS} -DDEBUG -Werror"
-		DEBUG_CFLAGS="-DDEBUG -Werror"
+		KERNELCPPFLAGS="${KERNELCPPFLAGS} -DDEBUG"
+		DEBUG_CFLAGS="-DDEBUG"
 		DEBUG_SPL="_with_debug"
 	], [
 		KERNELCPPFLAGS="${KERNELCPPFLAGS} -DNDEBUG"
@@ -556,7 +556,7 @@
 	modpost_flag=''
 	test "x$enable_linux_builtin" = xyes && modpost_flag='modpost=true' # fake modpost stage
 	AS_IF(
-		[AC_TRY_COMMAND(cp conftest.c build && make [$2] -C $LINUX_OBJ EXTRA_CFLAGS="-Werror-implicit-function-declaration $EXTRA_KCFLAGS" $ARCH_UM M=$PWD/build $modpost_flag) >/dev/null && AC_TRY_COMMAND([$3])],
+		[AC_TRY_COMMAND(cp conftest.c build && make [$2] -C $LINUX_OBJ EXTRA_CFLAGS="$EXTRA_KCFLAGS" $ARCH_UM M=$PWD/build $modpost_flag) >/dev/null && AC_TRY_COMMAND([$3])],
 		[$4],
 		[_AC_MSG_LOG_CONFTEST m4_ifvaln([$5],[$5])]
 	)
