diff -ruN a/config/spl-build.m4 b/config/spl-build.m4
--- a/config/spl-build.m4	2018-01-15 10:50:51.964146890 +0000
+++ b/config/spl-build.m4	2018-01-15 10:55:12.380158936 +0000
@@ -447,8 +447,8 @@
 
 	AS_IF([test "x$enable_debug" = xyes],
 	[
-		KERNELCPPFLAGS="${KERNELCPPFLAGS} -DDEBUG -Werror"
-		DEBUG_CFLAGS="-DDEBUG -Werror"
+		KERNELCPPFLAGS="${KERNELCPPFLAGS} -DDEBUG"
+		DEBUG_CFLAGS="-DDEBUG"
 		DEBUG_SPL="_with_debug"
 	], [
 		KERNELCPPFLAGS="${KERNELCPPFLAGS} -DNDEBUG"
@@ -555,7 +555,7 @@
 	modpost_flag=''
 	test "x$enable_linux_builtin" = xyes && modpost_flag='modpost=true' # fake modpost stage
 	AS_IF(
-		[AC_TRY_COMMAND(cp conftest.c build && make [$2] -C $LINUX_OBJ EXTRA_CFLAGS="-Werror-implicit-function-declaration $EXTRA_KCFLAGS" $ARCH_UM M=$PWD/build $modpost_flag) >/dev/null && AC_TRY_COMMAND([$3])],
+		[AC_TRY_COMMAND(cp conftest.c build && make [$2] -C $LINUX_OBJ EXTRA_CFLAGS="$EXTRA_KCFLAGS" $ARCH_UM M=$PWD/build $modpost_flag) >/dev/null && AC_TRY_COMMAND([$3])],
 		[$4],
 		[_AC_MSG_LOG_CONFTEST m4_ifvaln([$5],[$5])]
 	)
@@ -756,7 +756,7 @@
 
 AC_DEFUN([SPL_AC_SHRINKER_CALLBACK],[
 	tmp_flags="$EXTRA_KCFLAGS"
-	EXTRA_KCFLAGS="-Werror"
+	EXTRA_KCFLAGS=""
 	dnl #
 	dnl # 2.6.23 to 2.6.34 API change
 	dnl # ->shrink(int nr_to_scan, gfp_t gfp_mask)
@@ -901,7 +901,7 @@
 dnl #
 AC_DEFUN([SPL_AC_SET_FS_PWD_WITH_CONST],
 	tmp_flags="$EXTRA_KCFLAGS"
-	EXTRA_KCFLAGS="-Werror"
+	EXTRA_KCFLAGS=""
 	[AC_MSG_CHECKING([whether set_fs_pwd() requires const struct path *])
 	SPL_LINUX_TRY_COMPILE([
 		#include <linux/spinlock.h>
@@ -1042,7 +1042,7 @@
 AC_DEFUN([SPL_AC_FS_STRUCT_SPINLOCK], [
 	AC_MSG_CHECKING([whether struct fs_struct uses spinlock_t])
 	tmp_flags="$EXTRA_KCFLAGS"
-	EXTRA_KCFLAGS="-Werror"
+	EXTRA_KCFLAGS=""
 	SPL_LINUX_TRY_COMPILE([
 		#include <linux/sched.h>
 		#include <linux/fs_struct.h>
@@ -1311,7 +1311,7 @@
 AC_DEFUN([SPL_AC_RWSEM_SPINLOCK_IS_RAW], [
 	AC_MSG_CHECKING([whether struct rw_semaphore member wait_lock is raw])
 	tmp_flags="$EXTRA_KCFLAGS"
-	EXTRA_KCFLAGS="-Werror"
+	EXTRA_KCFLAGS=""
 	SPL_LINUX_TRY_COMPILE([
 		#include <linux/rwsem.h>
 	],[
@@ -1337,7 +1337,7 @@
 AC_DEFUN([SPL_AC_RWSEM_ACTIVITY], [
 	AC_MSG_CHECKING([whether struct rw_semaphore has member activity])
 	tmp_flags="$EXTRA_KCFLAGS"
-	EXTRA_KCFLAGS="-Werror"
+	EXTRA_KCFLAGS=""
 	SPL_LINUX_TRY_COMPILE([
 		#include <linux/rwsem.h>
 	],[
@@ -1362,7 +1362,7 @@
 	AC_MSG_CHECKING(
 	[whether struct rw_semaphore has atomic_long_t member count])
 	tmp_flags="$EXTRA_KCFLAGS"
-	EXTRA_KCFLAGS="-Werror"
+	EXTRA_KCFLAGS=""
 	SPL_LINUX_TRY_COMPILE([
 		#include <linux/rwsem.h>
 	],[
@@ -1572,7 +1572,7 @@
 AC_DEFUN([SPL_AC_INODE_LOCK], [
 	AC_MSG_CHECKING([whether inode_lock_shared() exists])
 	tmp_flags="$EXTRA_KCFLAGS"
-	EXTRA_KCFLAGS="-Werror"
+	EXTRA_KCFLAGS=""
 	SPL_LINUX_TRY_COMPILE([
 		#include <linux/fs.h>
 	],[
@@ -1594,7 +1594,7 @@
 AC_DEFUN([SPL_AC_GROUP_INFO_GID], [
 	AC_MSG_CHECKING([whether group_info->gid exists])
 	tmp_flags="$EXTRA_KCFLAGS"
-	EXTRA_KCFLAGS="-Werror"
+	EXTRA_KCFLAGS=""
 	SPL_LINUX_TRY_COMPILE([
 		#include <linux/cred.h>
 	],[
@@ -1617,7 +1617,7 @@
 AC_DEFUN([SPL_AC_KMEM_CACHE_CREATE_USERCOPY], [
 	AC_MSG_CHECKING([whether kmem_cache_create_usercopy() exists])
 	tmp_flags="$EXTRA_KCFLAGS"
-	EXTRA_KCFLAGS="-Werror"
+	EXTRA_KCFLAGS=""
 	SPL_LINUX_TRY_COMPILE([
 		#include <linux/slab.h>
 		static void ctor(void *foo)
@@ -1706,7 +1706,7 @@
 AC_DEFUN([SPL_AC_KERNEL_WRITE], [
 	AC_MSG_CHECKING([whether kernel_write() takes loff_t pointer])
 	tmp_flags="$EXTRA_KCFLAGS"
-	EXTRA_KCFLAGS="-Werror"
+	EXTRA_KCFLAGS=""
 	SPL_LINUX_TRY_COMPILE([
 		#include <linux/fs.h>
 	],[
@@ -1735,7 +1735,7 @@
 AC_DEFUN([SPL_AC_KERNEL_READ], [
 	AC_MSG_CHECKING([whether kernel_read() takes loff_t pointer])
 	tmp_flags="$EXTRA_KCFLAGS"
-	EXTRA_KCFLAGS="-Werror"
+	EXTRA_KCFLAGS=""
 	SPL_LINUX_TRY_COMPILE([
 		#include <linux/fs.h>
 	],[
